apps/emqx/src/emqx_cm.erl

```diff
@@ -455,60 +465,62 @@ discard_session(ClientId) when is_binary(ClientId) ->
 when
     Action :: kick | discard | {takeover, 'begin'} | {takeover, 'end'} | takeover_kick.
 request_stepdown(Action, ConnMod, Pid) ->
-    Timeout =
-        case Action == kick orelse Action == discard of
-            true -> ?T_KICK;
-            _ -> ?T_TAKEOVER
-        end,
+    Timeout = ?IF_KILL_ACTION(Action, ?T_KICK, ?T_TAKEOVER),
```

Comment:
maybe send a followup PR for 5.9 to change T_TAKEOVER from 15s to 5s
clients usually gives up waiting and reconnect before takeover can complete.

Reply:
Reply `CONNACK` quickly and then do the heavy tasks (taking over or discard session) would solve such kind of problem.

The 5s won't be enough for taking over a remove session, and the user may want to use an even shorter timeout when connection to emqx.

BTW. The default connection timeout of the paho mqtt java client is 30s, for both MQTT 5.0 and 3.1.1 clients.

Reply:
https://emqx.atlassian.net/browse/EMQX-13354